### PR TITLE
UIDATIMP-386: Add dropdown list with incoming record types to the RecordTypeSelector

### PR DESCRIPTION
**Pull request by mariia-aloshyna**
## Description

- Add a dropdown list with incoming record types to the left section of the graphic.
- Change labels of the Incoming record section and Incoming record field section depending on which Incoming record type is selected.
- Connect it to the form.
## Approach
- Render the `<Dropdown>` component instead of static `<div>` when the `<RecordTypeSelector>` component is editable.
- Change the `DropdownTrigger` label when a dropdown option changes.
- Add callbacks for the `<DropdownMenu>` onClick event:
  - Add selected incoming record type to the form values;
  - Change labels of the Incoming record section and Incoming record field section.

## Issues
[UIDATIMP-386](https://issues.folio.org/browse/UIDATIMP-386)
[UIDATIMP-351](https://issues.folio.org/browse/UIDATIMP-351)
## Screenshot
Dropdown

### CODE REVIEW

**Comment by taras-tkachenko** [comment]
Please align dropdown menu list to the left dropdown button border.

File: src/components/RecordTypesSelect/components/RecordItem.js [hunk 1]
```diff
@@ -1,56 +1,111 @@
...
 RecordItem.propTypes = {
   item: PropTypes.object.isRequired,
   className: PropTypes.string,
   style: PropTypes.object,
   onClick: PropTypes.func,
+  isEditable: PropTypes.bool,
+};
+
+RecordItem.defaultProps = {
+  onClick: noop,
```
Comment (Vladyslav-Velytskyi) [inline src/components/RecordTypesSelect/components/RecordItem.js]: It is unnecessary prop, cause in ` src/components/RecordTypesSelect/components/CompareRecordSelect.js` in this component you throw either function or noop.
Please recheck can this prop be omit

File: src/components/RecordTypesSelect/components/RecordSelect.js [hunk 2]
```diff
@@ -77,4 +79,10 @@ RecordSelect.propTypes = {
   onSelect: PropTypes.func,
   container: PropTypes.oneOfType([PropTypes.string, PropTypes.instanceOf(Element)]),
   treeData: PropTypes.object,
+  isEditable: PropTypes.bool,
+};
+
+RecordSelect.defaultProps = {
+  onSelect: noop,
```
Comment (Vladyslav-Velytskyi) [inline src/components/RecordTypesSelect/components/RecordSelect.js]: Same. this prop could be omit, you can make it required

**Comment by mariia-aloshyna** [comment]
> Please align dropdown menu list to the left dropdown button border.

@taras-tkachenko Done.

File: src/settings/MatchProfiles/MatchProfilesForm.js [hunk 3]
```diff
@@ -103,16 +106,43 @@ export const MatchProfilesFormComponent = memo(({
     dispatch(change(formName, 'profile.matchDetails[0].existingRecordType', type));
   };
 
+  const onIncomingRecordChange = record => {
+    setIncomingRecord(record);
+    dispatch(change(formName, 'profile.incomingRecordType', record.type));
+    dispatch(change(formName, 'profile.matchDetails[0].incomingRecordType', record.type));
```
[the remark this answers is not in the document]
Comment (mariia-aloshyna) [reply src/settings/MatchProfiles/MatchProfilesForm.js]: So each `matchDetails` record has the same incoming record type value? I mean we don't need to identify which of the `matchDetails` records should be updated?

File: src/components/RecordTypesSelect/RecordTypesSelect.js [hunk 4]
```diff
@@ -24,24 +31,29 @@ const useUpdateOnResize = () => {
...
   useUpdateOnResize();
   const [existingRecord, setExistingRecord] = useState(undefined);
+  const [incomingRecord, setIncomingRecord] = useState(undefined);
 
   useEffect(() => {
     setExistingRecord(FOLIO_RECORD_TYPES[existingRecordType] || undefined);
   }, [existingRecordType]);
 
+  useEffect(() => {
+    setIncomingRecord(INCOMING_RECORD_TYPES[incomingRecordType] || undefined);
```
Comment (Vladyslav-Velytskyi) [inline src/components/RecordTypesSelect/RecordTypesSelect.js]: I think you can rewrite it to `setIncomingRecord(INCOMING_RECORD_TYPES?.[incomingRecordType])` and it must works. This does not have to be changed, it is just a suggestion.

Comment (mariia-aloshyna) [reply src/components/RecordTypesSelect/RecordTypesSelect.js]: Thanks, @Vladyslav-Velytskyi! Rewrote.

**Pull request merged**